src/sql/workbench/contrib/query/browser/queryActions.ts

```diff
@@ -287,6 +289,30 @@ export class CancelQueryAction extends QueryTaskbarAction {
 	}
 }
 
+/**
+ * Telemetry helper class for query plan related actions.
+ */
+export class QueryPlanTelemetryHelper {
+	constructor(
+		@IConnectionManagementService private connectionService: IConnectionManagementService,
+		@IAdsTelemetryService private adsTelemetryService: IAdsTelemetryService
+	) { }
+
+	public addTelemetry(eventName: string, ownerUri: string, runOptions?: ExecutionPlanOptions): void {
+		const providerId: string = this.connectionService.getProviderIdFromUri(ownerUri);
```

Comment:
You should wrap all this in a try/catch. It's unlikely to fail, but we never want stuff to fail because something happened to telemetry. (the actual create/send event calls already have that on the telemetry service side. This would be for the other calls such as getProviderIdFromUri)